### ISSUE TITLE
Test Generated Clients & Document Changes

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#143]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 09:47 GMT+2:***

ACs:

* Changes to the generated clients have to be documented in the next release
  * TypeScript :white_check_mark:
  * Python :white_check_mark:
  * Java :white_check_mark:
  * Cpp :white_check_mark:
* The generated clients should be tested (and the testing documented) to avoid changes being unnoticed
  * TypeScript :white_check_mark:
  * Python :white_check_mark:
  * Java:white_check_mark:
  * Cpp:white_check_mark:
* Optional: Execute clients tests in the pipeline
  * TypeScript :white_check_mark:
  * Python :white_check_mark:
  * Java --\> won't do
  * Cpp --\> won't do

Hints:

* TypeScript is already used by our frontend
* CPP client is used by some colleagues
* Java client is used by shepard Timeseries Collector and shepard Process Wizard
* Python client is used by [sample data scripts](https://gitlab.com/dlr-shepard/shepard/-/blob/main/scripts/shepard_scripts/scripts/example_data.py?ref_type=heads)

**Assignees:** xit-paul-reinhardt

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/143*

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 09:47 GMT+2:***

added to epic &5

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 09:52 GMT+2:***

mentioned in task #136

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 12, 2024, 09:53 GMT+2:***

marked this issue as related to #136

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 14, 2024, 09:26 GMT+2:***

Hint: Maybe focus on the python client

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 09:57 GMT+2:***

mentioned in commit f9357bee2c6e6fa72a9679e0f610ac3967879d86

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 11:39 GMT+2:***

mentioned in commit 4ad9ade65ad924c299df946a5860f1e2544fb4e1

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 13:46 GMT+2:***

mentioned in commit 9e05fef79020ff5c67c418461c83fc514c105365

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 13:46 GMT+2:***

mentioned in commit f6512efebc92c28f68786e22d4eeeb2f532578ac

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 13:46 GMT+2:***

mentioned in commit 4d0f0b870089375d3ee3c708ca624d01c1924a14

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 13:47 GMT+2:***

mentioned in commit f173b34b5fcdea96bc1b58c61cd7f7b6e089f683

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:05 GMT+2:***

mentioned in commit c98ec4926e4b109123a6fa224ae96987b55b058f

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:09 GMT+2:***

mentioned in commit 0c3698dd3ce0244628f81210ae9d30c831aa691f

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:10 GMT+2:***

mentioned in commit 498f7f04d69afef6e33e42380e3e9d41bd7ed6dd

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:10 GMT+2:***

mentioned in commit 9162aec2a313c133f3148e5a53c8974c319030a8

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:33 GMT+2:***

mentioned in commit 9e10604083a96fe13faecf69b82a98c59bff896f

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:33 GMT+2:***

mentioned in commit f89b60b7c6ec298aa2f810b32268171f2d2c937a

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:37 GMT+2:***

mentioned in commit 4ceab3f2da0191edb4166acd508589e55d9a765e

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 14:39 GMT+2:***

mentioned in commit e305fc86af14feabaf37e8e273e0c14c15785373

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 15:13 GMT+2:***

mentioned in commit 369cff4ff4b5d1b083e4998c17ddb657d08f4d43

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 15, 2024, 15:33 GMT+2:***

mentioned in commit 4ed4a88468334d5262047b41426014f5ae5ac234

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 16, 2024, 10:55 GMT+2:***

mentioned in commit bb5b2aba967cad49c37a86ba916dd931eec03c95

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Aug 20, 2024, 10:12 GMT+2:***

mentioned in commit de2d9b1ca05382db526e45be226a7654adca5a62

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit f2ce7de44abc8b588185e29b16dac68969208cac

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 027583b10735176ff12a41a555c13f0e7aa8cc61

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 23d8bfe89a240a898e520dc34c305277624f6485

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 2ff154c0290b8444bcd3839fa7efe5d937e196bb

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit efe7705ee0e4716ee379b771e029c5abf80864f2

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit f14967a3e6d9da8f483be6aaee613b56f3b74cbf

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 6955947d5a4c556225fd25b906209ccc6b5a5fa8

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit eed541d5f078f19c470ef72f0ba5770a4f4594c3

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit ce0ad5e4bc9c65c915445b93db9e7497be74c003

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit b45e558f6e7657c0789a308aade07459822c4da7

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 363fc96a82d592f77dcafc7bdc02ea175574721d

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 9e91d15220ab74f5c5435bfd9782c591fd568d70

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit af9efc46c621d9c0f5993855aea40322676c43a8

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 619ba29dac0020c8e31fe6b4f3f3c0c876623168

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit d1710ee87a605ae9b1023a89095ae9c8fcded88c

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit eefc07b616f7bee1049bd0acd46243e2646557bb

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 20, 2024, 13:45 GMT+2:***

mentioned in commit 70d1a90b0b89ee2956a41262d419e9d03276969a

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 21, 2024, 10:18 GMT+2:***

Changes in java:

1. `QueryTypeEnum` becomes `QueryType`
2. QueryTypeEnum.**DATAOBJECT** becomes QueryType.**DATA_OBJECT**
3. `dataObjectApi.getAllDataObjects` parameters are still unordered :warning:
    - Fixed here: https://gitlab.com/dlr-shepard/shepard/-/merge_requests/215
4. `TraversalRulesEnum` becomes `TraversalRules`

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 23, 2024, 14:42 GMT+2:***

Last working cpp client: https://gitlab.com/dlr-shepard/backend/-/package_files/118260591/download

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Aug 23, 2024, 16:59 GMT+2:***

Just a reminder to maybe change the description of this issue (at the latest when it is closed).
Currently we publicly mention the name of the person that uses the CPP client and I don't think that this is a good idea.

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 26, 2024, 08:32 GMT+2:***

I took the enum notes over to the merge request description :)

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 26, 2024, 09:38 GMT+2:***

marked this issue as blocked by #186

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 28, 2024, 08:54 GMT+2:***

mentioned in commit 8bb69a1ff943facacf48ee77f8141e8f4d9b8f82

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 28, 2024, 08:57 GMT+2:***

mentioned in commit 9801931bf57519b0922b9556fb0a0904285bfac1

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Aug 28, 2024, 16:26 GMT+2:***

mentioned in commit 901f2320b5867fa0fff49f0dd52e05c231ad0704
